### PR TITLE
Fix klocwork issues in C-API integration branch

### PR DESCRIPTION
**Pull request by dkalinowski**

### CODE REVIEW

File: src/statefulmodelinstance.cpp [hunk 1]
```diff
@@ -238,6 +238,10 @@ Status StatefulRequestProcessor<tensorflow::serving::PredictRequest, tensorflow:
         }
     } else {
         auto modelState = inferRequest.query_state();
+        if (!sequence) {
+            SPDLOG_ERROR("sequence is not set");
```
Comment (atobiszei) [inline src/statefulmodelinstance.cpp:242]:
```suggestion
            SPDLOG_DEBUG("sequence is not set");
```

File: src/statefulmodelinstance.cpp [hunk 2]
```diff
@@ -254,7 +258,11 @@ Status StatefulRequestProcessor<tensorflow::serving::PredictRequest, tensorflow:
     Status status;
     if (sequenceProcessingSpec.getSequenceControlInput() == SEQUENCE_END) {
         sequenceManagerLock->lock();
-        status = sequenceManager.removeSequence(this->sequenceId);
+        if (!this->sequenceId.has_value()) {
+            SPDLOG_ERROR("sequenceId is not set");
```
Comment (atobiszei) [inline src/statefulmodelinstance.cpp:262]:
```suggestion
            SPDLOG_DEBUG("sequenceId is not set");
```

**Review by rasapala — APPROVED** [review]
LGTM

File: src/statefulmodelinstance.cpp (hunk 1 above)
Comment (dkalinowski) [reply src/statefulmodelinstance.cpp]: fixed

File: src/statefulmodelinstance.cpp (hunk 2 above)
Comment (dkalinowski) [reply src/statefulmodelinstance.cpp]: fixed

**Pull request merged**